js/web/lib/wasm/jsep/webgpu/ops/depth-to-space.ts

```diff
@@ -0,0 +1,111 @@
+    var a: ${input.type.indices};`);
+  for (let i = 0; i < rank; ++i) {
+    reverseFunc.push(input.indicesSet('a', perm[i], `i[${i}]`));
+  }
+  reverseFunc.push('return a;}');
+  return reverseFunc.join('\n');
+};
+
+const createDepthToSpaceProgramInfo = (inputTensor: TensorView, attributes: DepthToSpaceAttributes): ProgramInfo => {
+  let n: number, h: number, w: number, c: number;
+  let shape: number[];
+  let perm: number[];
+  const isChannelLast = attributes.format === 'NHWC';
+  const blocksize = attributes.blocksize;
+  const isDCRmode = attributes.mode === 'DCR';
+  if (isChannelLast) {
+    [n, h, w, c] = inputTensor.dims;
+    shape = isDCRmode ? [n, h, w, blocksize, blocksize, c / (blocksize ** 2)] :
+                        [n, h, w, c / (blocksize ** 2), blocksize, blocksize];
+    perm = isDCRmode ? [0, 1, 3, 2, 4, 5] : [0, 1, 4, 2, 5, 3];
+  } else {
+    [n, h, w, c] = [inputTensor.dims[0], inputTensor.dims[2], inputTensor.dims[3], inputTensor.dims[1]];
+    shape = isDCRmode ? [n, blocksize, blocksize, c / (blocksize ** 2), h, w] :
+                        [n, c / (blocksize ** 2), blocksize, blocksize, h, w];
+    perm = isDCRmode ? [0, 3, 4, 1, 5, 2] : [0, 1, 4, 2, 5, 3];
+  }
+  const reshapedInputTensor = inputTensor.reshape(shape);
+
+  const shapeBeforePerm = reshapedInputTensor.dims;
+  const shapeAfterPerm = ShapeUtil.sortBasedOnPerm(shapeBeforePerm, perm);
+
+  const inputDataType = inputTensor.dataType;
+  const reshapedInput = inputVariable('a', inputDataType, shapeBeforePerm);
+  const permedOutput = outputVariable('output', inputDataType, shapeAfterPerm);
+
+  const reshapedInputRank = reshapedInputTensor.dims.length;
+  const getShaderSource = (shaderHelper: ShaderHelper) => `
+  ${shaderHelper.registerUniform('output_size', 'u32').declareVariables(reshapedInput, permedOutput)}
+
+  ${permFunctionBody(perm, reshapedInputRank, reshapedInput, permedOutput)}
+
+  ${shaderHelper.mainStart()}
+    ${shaderHelper.guardAgainstOutOfBoundsWorkgroupSizes('uniforms.output_size')}
+
+    let indices = ${permedOutput.offsetToIndices('global_idx')};
+    let aIndices = perm(indices);
+
+    ${permedOutput.setByOffset('global_idx', reshapedInput.getByIndices('aIndices'))}
+  }`;
+
+  return {
+    name: 'DepthToSpace',
+    shaderCache: {hint: `${inputTensor.dims};${attributes.blocksize};${attributes.mode}`, inputDependencies: ['rank']},
+    getRunData: (inputs) => {
+      const outputShape = isChannelLast ? [n, h * blocksize, w * blocksize, c / (blocksize ** 2)] :
+                                          [n, c / (blocksize ** 2), h * blocksize, w * blocksize];
+      const outputSize = ShapeUtil.size(outputShape);
+      return {
+        outputs: [{dims: outputShape, dataType: inputs[0].dataType}],
+        dispatchGroup: {x: Math.ceil(outputSize / 64 /* workgroup size */)},
+        programUniforms:
+            [{type: DataType.uint32, data: outputSize}, ...createTensorShapeVariables(inputs[0].dims, outputShape)],
```

Comment:
`createTensorShapeVariables()` should use logical shape (ie. shapeBeforePerm and shapeAfterPerm)

Reply:
Use logical shape in the additional commit.